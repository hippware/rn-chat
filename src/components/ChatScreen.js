// @flow

import React, {Component} from 'react';
import {View, ActivityIndicator, Keyboard, Text, InteractionManager, TouchableOpacity, Image, StyleSheet, FlatList} from 'react-native';
import moment from 'moment';
import Button from 'react-native-button';
import {autorun, observable} from 'mobx';
import {observer} from 'mobx-react/native';
import {Actions} from 'react-native-router-flux';

import Screen from './Screen';
import Avatar from './Avatar';
import Chat from '../model/Chat';
import Message from '../model/Message';
import {showImagePicker} from './ImagePicker';
import ChatBubble from './ChatBubble';
import ChatMessage from './ChatMessage';
import location from '../store/locationStore';
import message from '../store/messageStore';
import model from '../model/model';
import Notification from './Notification';
import AutoExpandingTextInput from './AutoExpandingTextInput';
import {colors} from '../constants';

const onAttach = (item) => {
  const chat: Chat = item || console.error('No Chat is defined');
  showImagePicker('Select Image', (source, response) => {
    message.sendMedia({
      file: source,
      width: response.width,
      height: response.height,
      size: response.size,
      to: chat.id,
    });
  });
};

const AttachButton = ({item}) =>
  (<Button containerStyle={styles.sendButton} onPress={() => onAttach(item)}>
    <Image source={require('../../images/iconAttach.png')} />
  </Button>);

type Props = {
  item: Object,
};

type State = {
  text: string,
  isLoadingEarlierMessages: boolean,
};

class ChatScreen extends Component {
  props: Props;
  state: State;
  @observable messages: Array<any> = [];
  @observable chat: Chat;
  mounted: boolean;
  handler: Function;

  static renderTitle = ({item}) =>
    (<View>
      {model.chats.get(item).participants.map((profile, ind) =>
        (<TouchableOpacity
          key={`${ind}${profile.user}touch`}
          onPress={() => {
            Actions.profileDetail({item: profile, title: profile.displayName});
          }}
        >
          <Avatar size={40} profile={profile} isDay={location.isDay} />
        </TouchableOpacity>),
      )}
    </View>);

  constructor(props) {
    super(props);
    this.state = {
      text: '',
      isLoadingEarlierMessages: false,
    };
  }

  componentWillMount() {
    Keyboard.addListener('keyboardWillShow', this.keyboardWillShow.bind(this));
    Keyboard.addListener('keyboardWillHide', this.keyboardWillHide.bind(this));
  }

  componentDidMount() {
    this.mounted = true;

    // console.log('& cdm', this.props, this.chat, this.handler);
    const {props} = this;
    if (props.item && !this.chat && !this.handler) {
      this.chat = model.chats.get(props.item);
      // console.log('& this.chat', this.chat);
      this.handler = autorun(() => {
        if (this.chat) {
          this.createDatasource();
        }
      });
      this.chat &&
        InteractionManager.runAfterInteractions(() => {
          this.onLoadEarlierMessages(this.chat);
        });
    }
  }

  componentWillUnmount() {
    this.mounted = false;
    Keyboard.removeListener('keyboardWillShow');
    Keyboard.removeListener('keyboardWillHide');
    if (this.handler) {
      this.handler();
      this.handler = null;
    }
  }

  onLoadEarlierMessages = async (target) => {
    const chat: Chat = target || model.chats.get(this.props.item);
    if (!this.state.isLoadingEarlierMessages && !chat.loaded && !chat.loading) {
      this.setState({isLoadingEarlierMessages: true});
      await message.loadMore(chat);
      this.setState({isLoadingEarlierMessages: false});
    }
  };

  onSend = () => {
    console.log('& onSend');
    if (!this.state.text.trim() || !model.connected) {
      return;
    }
    message.sendMessage({to: this.chat.id, body: this.state.text.trim()});
    this.setState({text: ''});
  };

  keyboardWillShow = (e) => {
    if (this.mounted) this.setState({height: e.endCoordinates.height});
  };

  keyboardWillHide = (e) => {
    if (this.mounted) this.setState({height: 0});
  };

  renderDate = (rowData = {}) => {
    let diffMessage = null;
    diffMessage = this.getPreviousMessage(rowData);

    if (this.props.renderCustomDate) {
      return this.props.renderCustomDate(rowData, diffMessage);
    }

    if (rowData.date instanceof Date) {
      if (diffMessage === null) {
        return (
          <Text style={[styles.date]}>
            {moment(rowData.date).calendar()}
          </Text>
        );
      } else if (diffMessage.date instanceof Date) {
        const diff = moment(rowData.date).diff(moment(diffMessage.date), 'minutes');
        if (diff > 5) {
          return (
            <Text style={[styles.date]}>
              {moment(rowData.date).calendar()}
            </Text>
          );
        }
      }
    }
    return null;
  };

  getPreviousMessage = (message) => {
    for (let i = 0; i < this.messages.length; i++) {
      if (message.uniqueId === this.messages[i].uniqueId) {
        if (this.messages.length > i + 1) {
          return this.messages[i + 1];
        }
      }
    }
    return null;
  };

  getNextMessage = (message) => {
    for (let i = 0; i < this.messages.length; i++) {
      if (message.uniqueId === this.messages[i].uniqueId) {
        if (this.messages[i - 1]) {
          return this.messages[i - 1];
        }
      }
    }
    return null;
  };

  createDatasource = () => {
    this.messages = this.chat.messages
      .map((el: Message) => ({
        uniqueId: el.id,
        text: el.body || '',
        isDay: location.isDay,
        title: el.from.displayName,
        media: el.media,
        size: 40,
        position: el.from.isOwn ? 'right' : 'left',
        status: '',
        name: el.from.isOwn ? '' : el.from.displayName,
        image: el.from.isOwn || !el.from.avatar || !el.from.avatar.source ? null : el.from.avatar.source,
        profile: el.from,
        imageView: Avatar,
        view: ChatBubble,
        date: new Date(el.time),
      }))
      .reverse();
  };

  render() {
    if (!this.props.item || !this.messages.length) {
      return <Screen isDay={location.isDay} />;
    }
    if (this.chat) {
      InteractionManager.runAfterInteractions(() => {
        message.readAll(this.chat);
      });
    }
    return (
      <Screen isDay={location.isDay}>
        <View style={styles.container}>
          <FlatList
            data={this.messages.map(x => x)}
            renderItem={({item}) =>
              (<View>
                {this.renderDate(item)}
                <ChatMessage
                  rowData={item}
                  onErrorButtonPress={this.props.onErrorButtonPress}
                  displayNames={this.props.displayNames}
                  displayNamesInsideBubble={this.props.displayNamesInsideBubble}
                  diffMessage={this.getPreviousMessage(item)}
                  position={item.position}
                  forceRenderImage={this.props.forceRenderImage}
                  onImagePress={this.props.onImagePress}
                  onMessageLongPress={this.props.onMessageLongPress}
                  renderCustomText={this.props.renderCustomText}
                  parseText={this.props.parseText}
                  handlePhonePress={this.props.handlePhonePress}
                  handleUrlPress={this.props.handleUrlPress}
                  handleEmailPress={this.props.handleEmailPress}
                />
              </View>)}
            keyExtractor={item => item.uniqueId}
            onEndReached={this.onLoadEarlierMessages}
            onEndReachedThreshold={0.5}
            // ListFooterComponent={}
            // ListEmptyComponent={}
          />
          <View style={[styles.textInputContainer, location.isDay ? styles.textInputContainerDay : styles.textInputContainerNight]}>
            <AttachButton item={this.chat} />
            <AutoExpandingTextInput
              style={[styles.textInput, location.isDay ? styles.textInputDay : styles.textInputNight]}
              placeholder='Write a message'
              placeholderTextColor={colors.DARK_GREY}
              multiline
              autoFocus
              returnKeyType='default'
              onSubmitEditing={this.onSend}
              enablesReturnKeyAutomatically
              onChangeText={text => this.setState({text})}
              value={this.state.text}
              blurOnSubmit={false}
            />
            <TouchableOpacity style={styles.sendButton} onPress={this.onSend}>
              <Image source={!this.state.text.trim() || !model.connected ? require('../../images/iconSendInactive.png') : require('../../images/iconSendActive.png')} />
            </TouchableOpacity>
          </View>
          <View style={{height: this.state.height}} />
        </View>
        <Notification style={{position: 'absolute', top: 70}} />
      </Screen>
    );
  }
}

export default observer(ChatScreen);

const styles = StyleSheet.create({
  spiner: {
    flex: 1,
    padding: 20,
    backgroundColor: 'transparent',
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    flex: 1,
    paddingTop: 70,
    backgroundColor: 'transparent',
  },
  listView: {
    flex: 1,
  },
  textInputContainerDay: {
    backgroundColor: 'white',
  },
  textInputContainerNight: {
    backgroundColor: 'rgba(63,50,77,0.9)',
  },
  textInputContainer: {
    flexDirection: 'row',
    padding: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    alignSelf: 'center',
    width: 100,
    fontFamily: 'Roboto-Regular',
    flex: 1,
    margin: 0,
    padding: 0,
    paddingLeft: 20,
    paddingRight: 20,
    fontSize: 15,
  },
  textInputDay: {
    color: colors.DARK_PURPLE,
  },
  textInputNight: {
    color: 'white',
  },
  sendButton: {
    alignSelf: 'flex-end',
    paddingBottom: 5,
  },
  date: {
    color: '#aaaaaa',
    fontSize: 12,
    textAlign: 'center',
    fontWeight: 'bold',
    marginBottom: 8,
  },
  link: {
    color: '#007aff',
    textDecorationLine: 'underline',
  },
  linkLeft: {
    color: '#000',
  },
  linkRight: {
    color: '#fff',
  },
  loadEarlierMessages: {
    height: 44,
    justifyContent: 'center',
    alignItems: 'center',
  },
  loadEarlierMessagesButton: {
    fontSize: 14,
  },
});
