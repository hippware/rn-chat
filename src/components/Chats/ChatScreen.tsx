import React from 'react'
import {
  View,
  Keyboard,
  Text,
  TouchableOpacity,
  Image,
  StyleSheet,
  ActivityIndicator,
  FlatList,
} from 'react-native'

import moment from 'moment'
import {observable} from 'mobx'
import {observer, inject} from 'mobx-react/native'
import {isAlive} from 'mobx-state-tree'
import {Actions} from 'react-native-router-flux'
import Screen from '../Screen'
import {showImagePicker} from '../ImagePicker'
import ChatBubble from './ChatBubble'
import ChatMessage from './ChatMessage'
import {AutoExpandingTextInput, Avatar} from '../common'
import {colors} from '../../constants'
import {IWocky} from 'wocky-client'

const Button = require('apsl-react-native-button')

type Props = {
  item: string
  wocky?: IWocky
}

type State = {
  text: string
  height: number
}

const ChatTitle = inject('wocky')(
  observer(({item, wocky}) => {
    return wocky.chats.get(item)
      ? wocky.chats.get(item).participants.map((profile, ind) => (
          <TouchableOpacity
            key={`${ind}${profile.id}touch`} // eslint-disable-line
            onPress={() => {
              Actions.profileDetail({item: profile, title: profile.displayName})
            }}
          >
            <Avatar size={40} profile={profile} />
          </TouchableOpacity>
        ))
      : null
  })
)

@inject('wocky')
@observer
class ChatScreen extends React.Component<Props, State> {
  static renderTitle = ({item}) => <ChatTitle item={item} />

  state: State = {
    text: '',
    height: 0,
  }

  @observable messages: any[] = []
  @observable chat: any
  mounted: boolean = false
  handler: any
  list: any

  componentDidMount() {
    const {item, wocky} = this.props
    this.chat = wocky!.createChat(item)
    // load chat asynchronously
    this.chat.load().then(() => this.chat.readAll())
    this.chat.setActive(true)
    Keyboard.addListener('keyboardWillShow', this.keyboardWillShow)
    Keyboard.addListener('keyboardWillHide', this.keyboardWillHide)
    this.mounted = true
  }

  componentWillUnmount() {
    this.mounted = false
    if (this.chat) {
      this.chat.setActive(false)
    }
    Keyboard.removeListener('keyboardWillShow', this.keyboardWillShow)
    Keyboard.removeListener('keyboardWillHide', this.keyboardWillHide)
    if (this.handler) {
      this.handler()
      this.handler = null
    }
  }

  onSend = () => {
    if (this.chat.message.body.trim()) {
      this.chat.message.send()
    }
  }

  keyboardWillShow = e => {
    if (this.mounted) this.setState({height: e.endCoordinates.height})
  }

  keyboardWillHide = () => {
    if (this.mounted) this.setState({height: 0})
  }

  // TODO: rework this so it's included in row render...inefficient this way
  renderDate = (rowData: any = {}) => {
    let diffMessage: any = null
    diffMessage = this.getPreviousMessage(rowData)
    if (rowData.date instanceof Date) {
      if (diffMessage === null) {
        return <Text style={[styles.date]}>{moment(rowData.date).calendar()}</Text>
      } else if (diffMessage.date instanceof Date) {
        const diff = moment(rowData.date).diff(diffMessage.date, 'minutes')
        if (diff > 5) {
          return <Text style={[styles.date]}>{moment(rowData.date).calendar()}</Text>
        }
      }
    }
    return null
  }

  getPreviousMessage = message => {
    const i = this.messages.findIndex(m => m.uniqueId === message.uniqueId)
    return this.messages.length > i + 1 ? this.messages[i + 1] : null
  }

  renderItem = ({item}) =>
    item ? (
      <View>
        {this.renderDate(item)}
        <ChatMessage
          rowData={item}
          diffMessage={this.getPreviousMessage(item)}
          position={item.position}
        />
      </View>
    ) : null

  _footerComponent: any = observer(
    () =>
      this.chat && this.chat.loading ? <ActivityIndicator style={{marginVertical: 20}} /> : null
  )

  render() {
    return this.chat && isAlive(this.chat) ? (
      <Screen>
        <FlatList
          inverted
          data={this.chat.messages
            .map(el => {
              let media = null
              try {
                media = el.media
              } catch (err) {
                // console.log('TODO: fix Message.media reference error', err)
              }
              return el
                ? {
                    uniqueId: el.id,
                    text: el.body || '',
                    isDay: true,
                    title: el.from.displayName,
                    media,
                    size: 40,
                    position: el.from.isOwn ? 'right' : 'left',
                    status: '',
                    name: el.from.isOwn ? '' : el.from.displayName,
                    image:
                      el.from.isOwn || !el.from.avatar || !el.from.avatar.source
                        ? null
                        : el.from.avatar.source,
                    profile: el.from,
                    imageView: Avatar,
                    view: ChatBubble,
                    date: new Date(el.time),
                  }
                : null
            })
            .reverse()}
          ref={l => (this.list = l)}
          renderItem={this.renderItem}
          keyExtractor={i => i.uniqueId}
          onEndReached={() => this.chat.load()}
          onEndReachedThreshold={0.5}
          ListFooterComponent={this._footerComponent}
        />
        <InputArea chat={this.chat} onSend={this.onSend} />
        <View style={{height: this.state.height}} />
      </Screen>
    ) : (
      <Screen />
    )
  }
}

// separating text input here prevents unnecessary re-renders of the entire list when user enters text
const InputArea = inject('wocky')(
  observer(({wocky, chat, onSend}) => {
    return chat.message ? (
      <View style={[styles.textInputContainer, styles.textInputContainerDay]}>
        <AttachButton message={chat.message} />
        <AutoExpandingTextInput
          style={[styles.textInput, styles.textInputDay]}
          placeholder="Write a message"
          placeholderTextColor={colors.DARK_GREY}
          multiline
          autoFocus
          returnKeyType="default"
          enablesReturnKeyAutomatically
          onChangeText={t => chat.message && chat.message.setBody(t)}
          value={chat.message.body}
          blurOnSubmit={false}
          maxHeight={100}
          maxLength={500}
        />
        <TouchableOpacity disabled={!chat.message.body.trim() || !wocky.connected} onPress={onSend}>
          <Image
            source={
              chat.message.body.trim() && wocky.connected
                ? require('../../../images/iconSendActive.png')
                : require('../../../images/iconSendInactive.png')
            }
          />
        </TouchableOpacity>
      </View>
    ) : null
  })
)

const onAttach = (message, notificationStore) => {
  showImagePicker({
    title: 'Select Image',
    callback: async (source, response) => {
      try {
        await message.upload({
          file: source,
          width: response.width,
          height: response.height,
          size: response.size,
        })
        message.send()
      } catch (e) {
        notificationStore.flash(e.message)
      }
    },
  })
}

const AttachButton = inject('notificationStore')(({notificationStore, message}) => (
  <Button
    style={{borderWidth: 0, borderColor: 'transparent', paddingTop: 4}}
    onPress={() => onAttach(message, notificationStore)}
  >
    <Image source={require('../../../images/iconAttach.png')} />
  </Button>
))

export default ChatScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
  },
  listView: {
    flex: 1,
  },
  textInputContainerDay: {
    backgroundColor: 'white',
  },
  textInputContainerNight: {
    backgroundColor: 'rgba(63,50,77,0.9)',
  },
  textInputContainer: {
    flexDirection: 'row',
    paddingHorizontal: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    width: 100,
    fontFamily: 'Roboto-Regular',
    flex: 1,
    margin: 0,
    padding: 0,
    paddingLeft: 20,
    paddingRight: 20,
    fontSize: 15,
  },
  textInputDay: {
    color: colors.DARK_PURPLE,
  },
  textInputNight: {
    color: 'white',
  },
  date: {
    color: '#aaaaaa',
    fontSize: 12,
    textAlign: 'center',
    fontWeight: 'bold',
    marginBottom: 8,
  },
  link: {
    color: '#007aff',
    textDecorationLine: 'underline',
  },
  linkLeft: {
    color: '#000',
  },
  linkRight: {
    color: '#fff',
  },
  loadEarlierMessages: {
    height: 44,
    justifyContent: 'center',
    alignItems: 'center',
  },
  loadEarlierMessagesButton: {
    fontSize: 14,
  },
})
