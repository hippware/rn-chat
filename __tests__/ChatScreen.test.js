// @flow
/* global test */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ChatScreen from '../src/components/ChatScreen';
import {Provider} from 'mobx-react/native';
import {Chat} from 'wocky-client';

// mock autofocus: https://github.com/facebook/jest/issues/3707#issuecomment-311169259
jest.mock('TextInput', () => {
  const RealComponent = require.requireActual('TextInput');
  const React = require('React');
  class TextInput extends React.Component {
    render() {
      delete this.props.autoFocus;
      return React.createElement('TextInput', this.props, this.props.children);
    }
  }
  TextInput.propTypes = RealComponent.propTypes;
  return TextInput;
});

describe('ChatScreen', () => {
  test('renders with no data', () => {
    const wocky = {
      createChat: () => Chat.create({id: 'test'}),
      chats: {
        list: [],
        get: () => {
          return {
            id: '1234',
            setActive: () => {},
            messages: [],
          };
        },
      },
    };
    const tree = renderer
      .create(<Provider wocky={wocky}>
        <ChatScreen item='1234' />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  // TODO
  // test('renders with list of chats', () => {
  //   const wocky = {
  //     chats: {
  //       list: [],
  //     },
  //   };
  //   const tree = renderer.create(<ChatListScreen wocky={wocky} />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
