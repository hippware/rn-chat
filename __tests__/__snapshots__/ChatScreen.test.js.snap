// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen renders with no data 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <BVLinearGradient
    colors={
      Array [
        "rgb(255,255,255)",
        "rgb(241,242,244)",
        "rgb(243,244,246)",
      ]
    }
    locations={
      Array [
        0,
        0.2,
        1,
      ]
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <RCTScrollView
      ListFooterComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      inverted={true}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "scaleY": -1,
              },
            ],
          },
          undefined,
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scaleY": -1,
                },
              ],
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 20,
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
            "paddingTop": 4,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/iconAttach.png",
            }
          }
        />
      </View>
      <TextInput
        autoFocus={true}
        blurOnSubmit={false}
        enablesReturnKeyAutomatically={true}
        maxHeight={100}
        maxLength={500}
        multiline={true}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        placeholder="Write a message"
        placeholderTextColor="rgb(155,155,155)"
        returnKeyType="default"
        style={
          Array [
            Object {
              "flex": 1,
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "margin": 0,
              "padding": 0,
              "paddingLeft": 20,
              "paddingRight": 20,
              "width": 100,
            },
            Object {
              "color": "rgb(63,50,77)",
            },
          ]
        }
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/iconSendInactive.png",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    />
  </View>
</View>
`;
